ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -78,20 +73,23 @@ func (scbuilder *SiteConfigBuilder) getClusterCRs(clusterId int, siteConfigTemp
 			cluster := siteConfigTemp.Spec.Clusters[clusterId]
 			dataMap, err := scbuilder.getExtraManifest(dataMap, cluster)
 			if err != nil {
-				return clusterCRs, err
+				log.Printf("warning could not read extra-manifest %s.%s %s\n", cluster.ClusterName, cluster.ExtraManifestPath, err)
```

Comment:
If `getExtraManifest` returns an error, why are we continuing?  We don't have any guarantee that `dataMap` is still in a valid state for use below.

Reply:
cause the extra-manifest configMap if it has missing data doesn't prevent the cluster provisioning it will just raise warning. We don't validate the extra-manifest as we open it to end user

Reply:
This first level of extra-manifests is the one inside our container, which we consider critical to a successful deployment.  I agree for that the user-supplied additional directory that should be ignored.

Reply:
so if there is an error we should stop not continue or just log the error ? AND there is almost no chance that our pre-defined extra-manifest get errors when we add them. like we know the content and the files path (its hidden from end user) and we do test to make sure they r included correctly.

Reply:
Proper error-checking here also helps us make sure we haven't made a mistake, since we run this code in our CI lanes.

I'd say we should log and stop.

Reply:
Did this question get resolved? I think that any failure to fully generate what we expect + what the user asks for should result in an error. Otherwise the user is left to debug why their cluster is only partially functional or doesn't perform correctly. Issuing the error early and making it apparent to the user will shorten their iteration/test cycle.

Reply:
so I will change the message to error and return , then it will fail to generate the CRs and break in the main call